### PR TITLE
Documentation: Update documentation paragraphs related to unit tests

### PR DESCRIPTION
**Pull request by gziolo**
I updated our documentation to reflect our latest changes. I only touched parts directly related to our new single test runner. I'm sure we should spend more time in the near future refactoring React component unit testing document and rewrite it to promote `Enzyme`. We should also take another iteration on all documents once we drop `tests.json` files.

We should also migrate [this wiki page](https://github.com/Automattic/wp-calypso/wiki/Writing-and-Running-Tests) to docs folder and make it up to date :)

### CODE REVIEW

File: docs/our-approach-to-data.md [hunk 1]
```diff
@@ -97,7 +97,7 @@ client/state/
...
 
-For your tests to be included in the Redux state test runner, you must require them in `client/state/test/index.js`. Note that each `describe` block in this file should mirror the folder structure of state. If you're only interested in running tests for a single subtree of the global state, suffix the `make test` command with a dash followed by the name of the subtree. For example, to run the sites subtree tests, I would enter `make test-sites` in my Terminal.
+For your tests to be included in the Redux state test runner, you must (temporary) require them in `client/tests.json` ([learn how to do it](https://github.com/Automattic/wp-calypso/tree/master/test#extending-testsjson)). If you're only interested in running tests for a single subtree of the global state, use grep option in the `npm run test-client`. For example, to run the sites subtree tests, I would enter `npm run test-client -- --grep "sites"` in my Terminal.
```
Comment (aduth) [inline docs/our-approach-to-data.md]: I think we ought to drop the temporary note and simply circle back to remove this from the docs once it's changed. The test documentation is pretty thorough in how to add and run single tests, so perhaps we can reword this to indicate that:

>For your tests to be included in the Redux state test runner, you must add them to `client/tests.json`. The [tests documentation](https://github.com/Automattic/wp-calypso/tree/master/test#readme) includes information about how to include tests in the test runner, and instructions for running a single test file.

File: docs/react-component-unit-testing.md [hunk 2]
```diff
@@ -4,27 +4,14 @@ Calypso has a lot of React UI components. (Try for example running `find -name *
...
-     @NODE_ENV=test NODE_PATH=test:../../ $(MOCHA) --compilers jsx:babel/register --reporter $(REPORTER)
-
-.PHONY: test
-```
-
-
-Put this next to your component in a file named Makefile. Then make a test folder next to it and place your tests there. Now your tests should be runnable both from your component's folder and from all folders above it.
 
+It's highly possible that your tests will require to have DOM environment configured to work properly. We provide test helper that does all work for you.
+Check [Use fake DOM](https://github.com/Automattic/wp-calypso/tree/master/test/test/helpers/use-fake-dom) helper documentation to learn how to simulate DOM existence.
```
Comment (aduth) [inline docs/react-component-unit-testing.md]: Nitpicking grammar:

>It's very possible that your tests will assume the existence of a browser environment to work properly. Refer to the [Use fake DOM](https://github.com/Automattic/wp-calypso/tree/master/test/test/helpers/use-fake-dom) test helper documentation to learn how to configure an emulated DOM for your test.

File: test/README.md [hunk 3]
```diff
@@ -42,11 +42,14 @@ state
 
 ### How to run single test runner
 
-We provide two single test runners because of different node path rules applied. They contain files located in:
+You can run all tests from root folder with `npm test` command.
```
Comment (aduth) [inline test/README.md]: Nitpicking - A couple "the"s for readability:

>You can run all tests from the root folder using the `npm test` command.

**Comment by aduth** [comment]
Few nitpicks, but nice to see some much-needed updates to these docs.

**Comment by blowery** [comment]
Thanks for doing this @gziolo !

**Comment by gziolo** [comment]
@aduth Thanks for all corrections! Everything updated :)

**Comment by aduth** [comment]
Looks good :+1:
